src/main/java/fallingleaves/fallingleaves/particle/FallingLeafParticle.java

```diff
@@ -63,12 +63,16 @@ public ParticleTextureSheet getType() {
 
     @Environment(EnvType.CLIENT)
     public static class DefaultFactory implements ParticleFactory<DefaultParticleType> {
+        public static DefaultFactory lastInstance;
+
         private final SpriteProvider provider;
         //private BlockState block = Blocks.OAK_LEAVES.getDefaultState();
 
         public DefaultFactory(SpriteProvider provider) {
             this.provider = provider;
             //this.block = block;
+
+            lastInstance = this;
```

Comment:
It appears only ever one `ParticleFactory` is created, so we don't need to worry about `lastInstance` changing.